mlos_bench/mlos_bench/tests/launcher_parse_args_test.py

```diff
@@ -67,23 +66,65 @@ def test_launcher_args_parse_1(config_paths: List[str]) -> None:
     if sys.platform == 'win32':
         # Some env tweaks for platform compatibility.
         environ['USER'] = environ['USERNAME']
+    launcher = Launcher(description=desc, argv=cli_args.split())
+    # Check the basic parent service
+    assert isinstance(launcher.service, SupportsConfigLoading)  # built-in
+    assert isinstance(launcher.service, SupportsLocalExec)      # built-in
+    return launcher
 
-    # This is part of the minimal required args by the Launcher.
-    env_conf_path = 'environments/mock/mock_env.jsonc'
+
+def test_launcher_args_parse_defaults(config_paths: List[str]) -> None:
+    """
+    Test that we get the defaults we expect when using minimal config arg examples.
+    """
+    cli_args = '--config-paths ' + ' '.join(config_paths) + \
+        f' --environment {ENV_CONF_PATH}' + \
+        ' --globals globals/global_test_config.jsonc'
+    launcher = _get_launcher(__name__, cli_args)
+    # Check that the first --globals file is loaded and $var expansion is handled.
+    assert launcher.global_config['experiment_id'] == 'MockExperiment'
+    assert launcher.global_config['testVmName'] == 'MockExperiment-vm'
+    # Check that secondary expansion also works.
+    assert launcher.global_config['testVnetName'] == 'MockExperiment-vm-vnet'
+    # Check that we can expand a $var in a config file that references an environment variable.
+    assert path_join(launcher.global_config["pathVarWithEnvVarRef"], abs_path=True) \
+        == path_join(os.getcwd(), "foo", abs_path=True)
+    assert launcher.global_config["varWithEnvVarRef"] == f'user:{getuser()}'
+    assert launcher.teardown    # defaults
+    # Check that the environment that got loaded looks to be of the right type.
+    env_config = launcher.config_loader.load_config(ENV_CONF_PATH, ConfigSchema.ENVIRONMENT)
+    assert env_config["class"] == "mlos_bench.environments.mock_env.MockEnv"
+    assert check_class_name(launcher.environment, env_config['class'])
```

Comment:
why not just
```suggestion
    assert isinstance(launcher.environment, MockEnv)
```

Reply:
Prefer the former version since it checks two things independently:
1. were the configs read correctly
2. did we instantiate the class correctly
Makes debugging a failed test easier.